### PR TITLE
Added foreign keys to tables regions,cities, districts and refactor 'tableParams' logic

### PR DESCRIPTION
**Pull request by KizerovDmitriy**
# GreenCityUBS PR
## Summary Of Changes 🔥

- removed unuseded endpoints, enum 'UkraineRegions', service and repository methods,also refactored tests.
- added two new tables 'cities' and 'districts'.
- added two entity 'City' and 'District'.
- deleted unuse classes 'CityDto' and 'DistrictDto'.
- added @OneToMany annotation to establish a unidirectional relationship between the Region and City entities.
- added three new colums to table 'address' to set relationship with regions,cities and districts.
- added two new repositories to retrive data with specific parameters.
- refactor 'RegionRepository'.
- added migrations to DB.
- refactoring buisnes logic in 'OrdersAdminsPageServiceImpl' to retrive a list of all regions, cities and districts in 'tableParams' endpoint.
- added the 'BaseEntityForEnAndUkNames' class with the @MappedSuperclass annotation to create a reusable base class for entities like Region, City, and District, which share common fields for storing names in both English and Ukrainian.

## Issue link

[#7391](https://github.com/ita-social-projects/GreenCity/issues/7391)

# PR Checklist Forms

_(to be filled out by PR submitter)_
- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells or duplication
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

**Review by Maryna-511750 — APPROVED** [review]

**Review by RockTF — APPROVED** [review]

**Review by df7734 — APPROVED** [review]

File: dao/src/main/java/greencity/entity/user/Region.java [hunk 1]
```diff
@@ -1,42 +1,41 @@
...
 
     @OneToMany(cascade = CascadeType.ALL, mappedBy = "region")
     private List<Location> locations;
 
-    @Column(name = "name_uk")
-    private String ukrName;
-
-    @Column(name = "name_en")
-    private String enName;
+    @OneToMany(cascade = CascadeType.ALL, mappedBy = "region", fetch = FetchType.LAZY)
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/entity/user/Region.java]: Are you sure we need `CascadeType.ALL` here?

File: dao/src/main/java/greencity/entity/user/locations/City.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+public class City extends BaseEntityForEnAndUkNames {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
+    @JoinColumn(name = "region_id", nullable = false)
+    private Region region;
+
+    @OneToMany(cascade = CascadeType.ALL, mappedBy = "city", fetch = FetchType.LAZY)
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/entity/user/locations/City.java]: same here

File: dao/src/main/java/greencity/entity/user/Region.java (hunk 1 above)
Comment (KizerovDmitriy) [reply dao/src/main/java/greencity/entity/user/Region.java]: you are right, in this case will be a good to change it to PERSIST type

File: dao/src/main/java/greencity/entity/user/locations/City.java (hunk 2 above)
Comment (KizerovDmitriy) [reply dao/src/main/java/greencity/entity/user/locations/City.java]: same here

**Review by RomaMocherniuk — APPROVED** [review]

**Pull request merged**